src/IO/S3RequestSettings.cpp

```diff
@@ -205,73 +205,65 @@ void S3RequestSettings::updateIfChanged(const S3RequestSettings & settings)
 
 void S3RequestSettings::validateUploadSettings()
 {
-    static constexpr size_t min_upload_part_size_limit = 5 * 1024 * 1024;
```

Comment:
Those constraints were strictly related to Amazon S3 implementation.
S3 API can be used with other cloud storages, so here I made the settings validation more common.